NewsApi/auth.py

```diff
@@ -1,53 +1,44 @@
 import operator
 
 from flask import request, make_response, jsonify
-from NewsApi.app import app, db
+from NewsApi.app import app, exception_handler
 from NewsApi import models
-
-
-@app.route('/register', methods=['POST'])
-def sign_up():
-    new_user = models.User(username=request.json['username'],
-                           email=request.json['email'])
-    if not new_user:
-        error = 'Username is required.'
-    elif not password:
-        error = 'Password is required.'
-    elif db.execute(
-            'SELECT id FROM user WHERE username = ?', (username,)
-    ).fetchone() is not None:
-        error = f"User {username} is already registered."
-    db.session.add(new_user)
-    db.session.commit()
-    return make_response(jsonify(response='OK'), 201)
-
-
-
-@app.route('/register', methods=('GET', 'POST'))
-def register():
-    if request.method == 'POST':
-        username = request.form['username']
-        password = request.form['password']
-        db = get_db()
-        error = None
-
-        if not username:
-            error = 'Username is required.'
-        elif not password:
-            error = 'Password is required.'
-        elif db.execute(
-            'SELECT id FROM user WHERE username = ?', (username,)
-        ).fetchone() is not None:
-            error = f"User {username} is already registered."
-
-        if error is None:
-            db.execute(
-                'INSERT INTO user (username, password) VALUES (?, ?)',
-                (username, generate_password_hash(password))
-            )
-            db.commit()
-            return redirect(url_for('auth.login'))
-
-        flash(error)
-
-    return render_template('auth/register.html')
\ No newline at end of file
+from flask_restful import Resource
+from flask_jwt_extended import create_access_token
+from NewsApi.errors import SchemaValidationError, EmailAlreadyExistsError, UserPermissionError, InternalServerError
+
+
+class UserRegistration:
```

Comment:
It does not look like this is used anywhere in the rest of the codebase. Are you in the middle of the changes to start using `flask_restful`? 

I would personally suggest you to stick to the project structure that we presented in class and focus on getting the DB structure right first. 

If you then want to experiment with some flask plugins that's also cool, but just make sure to make it after the *mandatory* parts of the projects are covered.

Reply:
Thank you @lucaspal, 

I sincerely appreciate your help with the project :). I created a new PR where I made some changes to the structure of the project and added some user management functionalities. I checked all the routes on Postman, they seem to work fine. Please have a look at it again and let me know if there are any issues. 

Once again, I appreciate the time and effort you put into this matter. 

PS. While you review my latest PR, my next will be testing.